### ISSUE TITLE
typing structuredContent indirectly causes type error

### ISSUE THREAD

**Issue by Autumnal-Joy** [modelcontextprotocol/typescript-sdk#837]
**Describe the bug**
`structuredContent` causes type error

**To Reproduce**
```typescript
interface SomeType {
  data: string;
}

const server = new McpServer({
  name: "MCP",
  version: "1.0.0",
});

server.registerTool(
  "name",
  {
    outputSchema: {
      data: z.string(),
    },
  },
  async () => { // show type error
    return {} as {
      content: [];
      structuredContent: SomeType; // <-- cause type error
      isError?: boolean;
    };
  }
);
```

If I **inline** the type of `structuredContent` within the code, it will not report a type error.
```typescript
const server = new McpServer({
  name: "MCP Refine",
  version: "1.0.0",
});

server.registerTool(
  "name",
  {
    outputSchema: {
      data: z.string(),
    },
  },
  async () => {
    return {} as {
      content: [];
      structuredContent:  { // <-- work fine
        data: string;
      };
      isError?: boolean;
    };
  }
);
```

**Expected behavior**
Do not cause the type error

**Logs**

**Additional context**
"@modelcontextprotocol/sdk": "1.17.1",
"zod": "3.25.76"

**Comment by ochafik** [comment]
Hi @Autumnal-Joy, thanks for reporting this!

The output type (`CallToolResult` through `ToolCallback`) isn't currently parameterized to the outputSchema, indeed. We'd need to make this work even when the outputSchema isn't set. A PR would be welcome :-)

**Comment by ochafik** [comment]
(also, might wanna do this as part of https://github.com/modelcontextprotocol/typescript-sdk/issues/911)

**Comment by NorthIsUp** [comment]
are there any examples/docs right now for current correct usage of structuredContent?

**Comment by felixweinberger** [comment]
This is a TypeScript structural-typing quirk: named interfaces don't get implicit index signatures, so they aren't assignable to `Record<string, unknown>` / `{ [key: string]: unknown }`.

Workarounds (verified against the SDK types):
- Use `type MyResult = {...}` instead of `interface MyResult {...}` (type aliases are assignable)
- Spread: `structuredContent: { ...result }`
- Add an explicit index signature: `interface MyResult { data: string; [key: string]: unknown }`

We'll add this to the docs. Closing as a TypeScript-level workaround rather than an SDK change, but happy to revisit if these don't cover a real use case.
